docs/content/concepts/partitions-schedules-sensors/sensors.mdx

```diff
@@ -205,6 +191,58 @@ def every_fifth_asset_c_sensor(context):
         return SkipReason(f"asset_c only materialized {len(asset_events)} times.")
 ```
 
+#### Triggering a partitioned run after upstream partitions are materialized
+
+The <PyObject object="MultiAssetSensorEvaluationContext" method="latest_materialization_records_by_partition"/> method fetches the latest materialization records by partition for a given asset. Using this method, you can trigger downstream partitioned runs when the corresponding partitions of upstream assets are materialized.
+
+The following example monitors two upstream daily-partitioned assets, kicking off materializations of the corresponding partition in the downstream daily-partitioned asset. Whenever a partition is replaced in `upstream_daily_1` and the same partition has previously been materialized in `upstream_daily_2` (and vice versa), the sensor yields a run request for the downstream asset.
+
+```python file=/concepts/partitions_schedules_sensors/sensors/sensors.py startafter=start_daily_partitioned_asset_with_two_upstreams endbefore=end_daily_partitioned_asset_with_two_upstreams
+@multi_asset_sensor(
+    asset_keys=[AssetKey("upstream_daily_1"), AssetKey("upstream_daily_2")],
+    job=downstream_daily_job,
+)
+def trigger_daily_asset_if_all_upstream_partitions_materialized(context):
+    run_requests = []
+
+    def evaluate_latest_partitions(asset_key):
```

Comment:
Nesting functions inside of other functions is kind of a "fancy" Python feature that might throw some users off. I imagine the idea here is to avoid the text getting squished with all the layers of indenting, but I'm not sure it's worth it.